naga/src/front/wgsl/lower/mod.rs

```diff
@@ -2142,7 +2142,27 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
             Some(&LoweredGlobalDecl::Function(function)) => {
                 let arguments = arguments
                     .iter()
-                    .map(|&arg| self.expression(arg, ctx))
+                    .enumerate()
+                    .map(|(i, &arg)| {
+                        // Try to convert abstract values to the known argument types
+                        match ctx.module.functions[function]
+                            .arguments
+                            .get(i)
+                            .map(|arg| arg.ty)
+                        {
+                            Some(arg_ty) => {
+                                let expr = self.expression_for_abstract(arg, ctx)?;
+                                ctx.try_automatic_conversions(
+                                    expr,
+                                    &crate::proc::TypeResolution::Handle(arg_ty),
+                                    ctx.ast_expressions.get_span(arg),
+                                )
+                            }
+                            // Wrong number of arguments... just concretize the type here
+                            // and let the validator report the error.
+                            None => self.expression(arg, ctx),
+                        }
+                    })
```

Comment:
todo: Ah, one thing: gotta `CHANGELOG` entry? 🙂 This sounds like a nice item to highlight there!

Reply:
done 🙂